src/domotica/domotica.c

```diff
@@ -22,48 +22,52 @@ __attribute__ ((weak, alias ("domotica_handle_output_change_dummy"))) \
 
 // ------------------------------------------------------------------
 // Internal structure of a queue to maintain all changes that should be handled
-typedef struct OUTPUT_CHANGE {
+typedef struct {
   uint16_t mask_on;
   uint16_t mask_off;
-  struct OUTPUT_CHANGE *next;
 } OUTPUT_CHANGE_Type;
 
-static OUTPUT_CHANGE_Type *output_change_queue = 0;
+typedef struct {
+  OUTPUT_CHANGE_Type buffer[DOMOTICA_CHANGE_BUFFER_Size];
+  volatile uint8_t writer;
+  volatile uint8_t reader;
+} OUTPUT_CHANGE_RINGBUFFER_Type;
```

Comment:
Big reason is that the OUTPUT_CHANGE_ are internal variables that represent the buffer (and are static with a comprehensive name), whereas DOMOTICA_CHANGE_BUFFER_Size can be externally defined. For naming I found this more intuitive than having DOMOTICA_OUTPUT_CHANGE_... as a name...